src/xrpld/consensus/Consensus.h

```diff
@@ -1039,41 +1077,63 @@ Consensus<Adaptor>::handleWrongLedger(typename Ledger_t::ID const& lgrId)
     }
 
     if (previousLedger_.id() == prevLedgerID_)
+    {
+        CLOG(clog) << "previousLedger_.id() == prevLeverID_ " << prevLedgerID_
+                   << ". ";
         return;
+    }
 
     // we need to switch the ledger we're working from
     if (auto newLedger = adaptor_.acquireLedger(prevLedgerID_))
     {
         JLOG(j_.info()) << "Have the consensus ledger " << prevLedgerID_;
+        CLOG(clog) << "Have the consensus ledger " << prevLedgerID_ << ". ";
         startRoundInternal(
-            now_, lgrId, *newLedger, ConsensusMode::switchedLedger);
+            now_, lgrId, *newLedger, ConsensusMode::switchedLedger, clog);
     }
     else
     {
+        CLOG(clog) << "Still on wrong ledger. ";
         mode_.set(ConsensusMode::wrongLedger, adaptor_);
     }
 }
 
 template <class Adaptor>
 void
-Consensus<Adaptor>::checkLedger()
+Consensus<Adaptor>::checkLedger(std::unique_ptr<std::stringstream> const& clog)
 {
+    CLOG(clog) << "checkLedger. ";
+
     auto netLgr =
         adaptor_.getPrevLedger(prevLedgerID_, previousLedger_, mode_.get());
+    CLOG(clog) << "network ledgerid " << netLgr << ",  " << "previous ledger "
+               << prevLedgerID_ << ". ";
 
     if (netLgr != prevLedgerID_)
     {
-        JLOG(j_.warn()) << "View of consensus changed during "
-                        << to_string(phase_) << " status=" << to_string(phase_)
-                        << ", " << " mode=" << to_string(mode_.get());
-        JLOG(j_.warn()) << prevLedgerID_ << " to " << netLgr;
+        std::stringstream ss;
+        ss << "View of consensus changed during " << to_string(phase_)
+           << " status=" << to_string(phase_) << ", "
+           << " mode=" << to_string(mode_.get());
+        JLOG(j_.warn()) << ss.str();
+        CLOG(clog) << ss.str();
+        ss.clear();
+        ss.str("");
+        ss << prevLedgerID_ << " to " << netLgr;
+        JLOG(j_.warn()) << ss.str();
+        CLOG(clog) << ss.str();
```

Comment:
By itself, this second message is pretty useless. Why not take this opportunity to combine the two?

Reply:
fixed